packages/core/test/utils_test.js

```diff
@@ -2540,6 +2540,282 @@ describe("utils", () => {
+                  Breed: {
+                    title: "Breed",
+                    properties: {
+                      BreedName: {
+                        default: "Alsatian",
+                        enum: ["Alsatian", "Dalmation"],
+                        title: "Breed name",
+                        type: "string",
+                      },
+                    },
+                    allOf: [
+                      {
+                        if: {
+                          required: ["BreedName"],
+                          properties: {
+                            BreedName: {
+                              const: "Alsatian",
+                            },
+                          },
+                        },
+                        then: {
+                          properties: {
+                            Fur: {
+                              default: "brown",
+                              enum: ["black", "brown"],
+                              title: "Fur",
+                              type: "string",
+                            },
+                          },
+                          required: ["Fur"],
+                        },
+                      },
+                      {
+                        if: {
+                          required: ["BreedName"],
+                          properties: {
+                            BreedName: {
+                              const: "Dalmation",
+                            },
+                          },
+                        },
+                        then: {
+                          properties: {
+                            Spots: {
+                              default: "small",
+                              enum: ["large", "small"],
+                              title: "Spots",
+                              type: "string",
+                            },
+                          },
+                          required: ["Spots"],
+                        },
+                      },
+                    ],
+                    required: ["BreedName"],
+                  },
+                },
+              },
+            },
+          ],
+          required: ["Animal"],
+        };
+        const definitions = {};
+        const formData = {
+          Animal: "Cat",
+          Tail: "Long",
+        };
+
+        expect(retrieveSchema(schema, { definitions }, formData)).eql({
+          type: "object",
+          properties: {
+            Animal: {
+              default: "Cat",
+              enum: ["Cat", "Dog"],
+              title: "Animal",
+              type: "string",
+            },
+            Tail: {
+              default: "Long",
+              enum: ["Long", "Short", "None"],
+              title: "Tail length",
+              type: "string",
+            },
+          },
+          required: ["Animal", "Tail"],
+        });
+      });
+      it("should resolve multiple conditions in nested allOf blocks", () => {
```

Comment:
Can we change the test names to capture what the difference is between the tests, and more importantly so we don't have two tests with the same name?

Reply:
Follow up question: are both tests necessary? Do they capture separate edge cases that we effectively handle differently?

Reply:
Thanks, you're right, we don't need both these tests, only the second one, which captures the failing behaviour in the current live code and where the new code addresses this issue. Removed.